app/models/sort_facet.rb

```diff
@@ -10,6 +12,14 @@ def name
   end
   alias_method :ga4_section, :name
 
+  def key
+    KEY
+  end
+
+  def options
```

Comment:
I'm not super-enthused about the idea of the model knowing about the presenter - I think it might make more sense if the presenter is something the view creates/calls passing the model to it.

Reply:
Yeah, me neither. Irritatingly most of the existing facet logic contains a lot of presentation concerns, so I felt less bad about doing this here.

The partial that renders this facet does implicitly have access to the sort presenter (by virtue of the controller action setting it up as an instance variable), so if you think it would be better to just use it from the view instead of passing it through here I'm happy to do that?

Reply:
Ah, a classic quandary: "should I follow the broader idiom, or should I follow the idiom of this specific app?".

I think in this case if it doesn't add in a lot of extra code it would be better to use it from the view, and you can remove the model's requirement to know anything about the presenter.